src/livecodes/languages/typescript/lang-typescript.ts

```diff
@@ -22,11 +24,14 @@ export const typescript: LanguageSpecs = {
     url: typescriptUrl,
     factory:
       () =>
-      async (code, { config, language }) =>
+      async (code, { config }) =>
         (window as any).ts.transpile(code, {
           ...typescriptOptions,
+          ...(['jsx', 'tsx'].includes(config.script.language) && !hasCustomJsxRuntime(code)
+            ? { jsx: 'react-jsx' }
+            : {}),
```

Comment:
let's keep the change more selective to avoid interference with other runtimes (e.g. Vue with JSX)